srf_generation/velocity_model_generation/q_model_generation/EP_to_bin.py

```diff
@@ -0,0 +1,51 @@
+"""Script to generate a Qp/Qs binary model file from an Eberhart-Phillips style ascii model"""
+import argparse
+from os.path import abspath
+import pandas as pd
+
+from qcore.vm_file import VelocityModelFile
+
+
+def load_args():
+    parser = argparse.ArgumentParser(allow_abbrev=False)
+    parser.add_argument("infile", type=abspath)
+    parser.add_argument("outfile", type=abspath)
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    args = load_args()
```

Comment:
I think this part belongs under `__main__`, not in the main function.To me main is a function like any other (`__main__` is the starting point in  a script), hence it should be re-usable and self-contained. So I usually do all the argument stuff under` __main__` and then pass the parsed arguments to main, i.e. main becomes
```
def main(infile: Path, outfile: Path):
    .....
```

Anyways just my 5 cents, in my opinion it makes the code cleaner, so I like it haha.

Reply:
My 2c is that I don't like args (and really anything unnecessary) as a global variable so returning args into __main__ gives it global scope.

I normally do mains like this, but only use it for processing arguments really and then hands off to a function that does the body of the script. So I'd only use main in the script that has it.

I definitely see why you would do it the way you do. I think we might have even had this conversation before at some point.

Reply:
I think Jason got to me first on `main()` vs `__main__` style, so I normally go with his way.
Using non constant globals and args outside main is out, `__name__=="__main__":main()` is in :stuck_out_tongue: